switchsdk/src/main/java/net/gini/switchsdk/TakePicturePresenter.java

```diff
@@ -94,16 +94,18 @@ public void onOrderCompleted(@NonNull final String extractionUrl) {
                 new ExtractionService.ExtractionListener() {
                     @Override
                     public void onExtractionsReceived() {
-                        //Check if the user is in the camera screen
-                        if (canExitSdk()) {
-                            int resultCode = SwitchSdk.EXTRACTIONS_AVAILABLE;
-                            exitSdkWithCleanup(resultCode);
-                        }
                     }
                 });
     }
 
     @Override
+    public void onPictureKept() {
+        if (mExtractionService.extractionsAvailable()) {
```

Comment:
I assume you checked and it's the right one, but `extractionsAvailable` is not an ideal name. There might be extractions, but not all of them. I guess, that's something for @EarlOfEgo

Reply:
Yes - `extractionsAvailable()` basically returns if there is a cached result from a `fetchExtractions` operation. The SDK only fetches extractions in two cases: the user presses the "finish" button or the extraction order has its `extractionsComplete` field set to `true`.